client/rpc/trace/src/lib.rs

```diff
@@ -901,24 +901,32 @@ where
 		let mut traces: Vec<_> =
 			moonbeam_client_evm_tracing::formatters::TraceFilter::format(proxy)
 				.ok_or("Fail to format proxy")?;
+
+		let eth_transactions_by_index: BTreeMap<u32, H256> = eth_transactions
+			.iter()
+			.map(|t| (t.transaction_index, t.transaction_hash))
+			.collect();
+
 		// Fill missing data.
-		for trace in traces.iter_mut() {
+		for trace in traces
+			.iter_mut()
+			.filter(|t| eth_transactions_by_index.contains_key(&t.transaction_position))
+		{
 			trace.block_hash = eth_block_hash;
 			trace.block_number = height;
-			trace.transaction_hash = eth_transactions
-				.get(trace.transaction_position as usize)
-				.ok_or_else(|| {
-					tracing::warn!(
-						"Bug: A transaction has been replayed while it shouldn't (in block {}).",
-						height
-					);
+			trace.transaction_hash =
+				match eth_transactions_by_index.get(&trace.transaction_position) {
+					Some(transaction_hash) => transaction_hash.clone(),
+					None => {
```

Comment:
This variant become unreachable, and you read the eth_transactions_by_index map twice, please apply my previous comment and remove this code